src/test/java/org/opensearch/flowframework/rest/FlowFrameworkRestApiIT.java

```diff
@@ -363,6 +363,281 @@ public void testCreateAndProvisionAgentFrameworkWorkflow() throws Exception {
+
+        resourcesCreated = getResourcesCreated(client(), workflowId, 10);
+        assertEquals(5, resourcesCreated.size());
+        resourceIds = resourcesCreated.stream().map(x -> x.workflowStepName()).collect(Collectors.toList());
+        assertTrue(resourceIds.contains("create_connector"));
+        assertTrue(resourceIds.contains("register_remote_model"));
+        assertTrue(resourceIds.contains("create_ingest_pipeline"));
+        assertTrue(resourceIds.contains("create_index"));
+
+        // Retrieve index settings to ensure pipeline ID is set correctly
+        String indexName = resourcesCreated.stream()
+            .filter(x -> x.workflowStepName().equals("create_index"))
+            .findFirst()
+            .get()
+            .resourceId();
+        Map<String, Object> indexSettings = getIndexSettingsAsMap(indexName);
+        assertEquals(pipelineId, indexSettings.get("index.default_pipeline"));
+
+        // Reprovision template to remove default ingest pipeline
+        template = TestHelpers.createTemplateFromFile("registerremotemodel-ingestpipeline-updateindex.json");
+        response = reprovisionWorkflow(client(), workflowId, template);
+        assertEquals(RestStatus.CREATED, TestHelpers.restStatus(response));
+
+        resourcesCreated = getResourcesCreated(client(), workflowId, 10);
+        // resource count should remain unchanged when updating an existing node
+        assertEquals(5, resourcesCreated.size());
+
+        // Retrieve index settings to ensure default pipeline has been updated correctly
+        indexSettings = getIndexSettingsAsMap(indexName);
+        assertEquals("_none", indexSettings.get("index.default_pipeline"));
+
+        // Deprovision and delete all resources
+        Response deprovisionResponse = deprovisionWorkflowWithAllowDelete(client(), workflowId, pipelineId + "," + indexName);
+        assertBusy(
+            () -> { getAndAssertWorkflowStatus(client(), workflowId, State.NOT_STARTED, ProvisioningProgress.NOT_STARTED); },
+            60,
+            TimeUnit.SECONDS
+        );
+        assertEquals(RestStatus.OK, TestHelpers.restStatus(deprovisionResponse));
+
+        // Hit Delete API
+        Response deleteResponse = deleteWorkflow(client(), workflowId);
+        assertEquals(RestStatus.OK, TestHelpers.restStatus(deleteResponse));
+    }
+
+    public void testReprovisionWorkflowMidNodeAddition() throws Exception {
+        // Begin with a template to register a local pretrained model and create an index, no edges
+        Template template = TestHelpers.createTemplateFromFile("registerremotemodel-createindex.json");
+
+        // Hit Create Workflow API to create agent-framework template, with template validation check and provision parameter
+        Response response;
+        if (!indexExistsWithAdminClient(".plugins-ml-config")) {
+            assertBusy(() -> assertTrue(indexExistsWithAdminClient(".plugins-ml-config")), 40, TimeUnit.SECONDS);
```

Comment:
See above comment about moving this into the `@Before` method, perhaps replacing what's currently there.